CDP4OfficeInfrastructure/OfficeDal/CrossviewWorkbookData.cs

```diff
@@ -0,0 +1,225 @@
+        /// Gets the <see cref="IEnumerable{ElementDefinition}"/> instances.
+        /// </summary>
+        [XmlIgnore]
+        public IEnumerable<Thing> ElementDefinitionList => this.Serializer.Deserialize(this.GenerateStreamFromString(this.ElementDefinitionData.Value));
+
+        /// <summary>
+        /// Gets the <see cref="IEnumerable{ParameterType}"/> instances.
+        /// </summary>
+        [XmlIgnore]
+        public IEnumerable<Thing> ParameterTypeList => this.Serializer.Deserialize(this.GenerateStreamFromString(this.ParameterTypeData.Value));
+
+        /// <summary>
+        /// Gets or sets a dictionary that contains cell names and cell values that has been modified
+        /// </summary>
+        [XmlIgnore]
+        public Dictionary<string, string> ManuallySavedValues
+        {
+            get
+            {
+                var result = new Dictionary<string, string>();
+
+                for(var i = 0; i < this.CellNamesData.Value.Split('|').Length; i++)
+                {
+                    result[this.CellNamesData.Value.Split('|')[i]] = this.CellValuesData.Value.Split('|')[i];
+                }
+
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CrossviewWorkbookData"/> class.
+        /// </summary>
+        public CrossviewWorkbookData()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CrossviewWorkbookData"/> class.
+        /// </summary>
+        /// <param name="elementDefinitions">
+        /// The element definitions list that needs to be preserved <see cref="IEnumerable{ElementDefinition}" />
+        /// </param>
+        /// <param name="parameterTypes">
+        /// The parameter types list that needs to be preserved <see cref="IEnumerable{ParameterType}" />
+        /// </param>
+        /// <param name="manuallySavedValues">
+        /// The cell name/value key/value pai that needs to be preserved <see cref="Dictionary{TKey,TValue}"/>
+        /// </param>
+        public CrossviewWorkbookData(IEnumerable<CDP4Common.EngineeringModelData.ElementDefinition> elementDefinitions, IEnumerable<CDP4Common.SiteDirectoryData.ParameterType> parameterTypes,
+            Dictionary<string, string> manuallySavedValues)
+        {
+            var preservedDefinitions = elementDefinitions.Select(elementDefinition => elementDefinition.ToDto() as ElementDefinition).ToList();
+            this.elementDefinitionData =  this.GenerateStringFromList(preservedDefinitions);
+
+            var preservedTypes = parameterTypes.Select(parameterType => parameterType.ToDto() as ParameterType).ToList();
+            this.parameterTypeData = this.GenerateStringFromList(preservedTypes);
+
+            this.cellNamesData = string.Join("|", manuallySavedValues.Keys.ToArray());
+
+            this.cellValuesData = string.Join("|", manuallySavedValues.Values.ToArray());
```

Comment:
We need to find a way so this serialization does not depend on the presence or absence of the hardcoded separator (which should be a constant at the top) inside the values.

For the values, this doesn't work, as values are allowed to contain the `|` character.

Reply:
Added a constant for the moment.

Reply:
I will try with XmlSerializer.